projects/Dispersion/Dispersion.cpp

```diff
@@ -103,55 +104,55 @@ namespace projects {
    void Dispersion::hook(
       cuint& stage,
       const dccrg::Dccrg<spatial_cell::SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
-      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH> & perBGrid
+      FsGrid<Real, fsgrids::bfield::N_BFIELD, FS_STENCIL_WIDTH> & perBGrid
    ) const {
       if(hook::END_OF_TIME_STEP == stage) {
          int myRank;
          MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
 
-         vector<Real> localRhom(P::xcells_ini, 0.0), outputRhom(P::xcells_ini, 0.0);
+         vector<Real> localRhom(meshParams.xcells_ini, 0.0), outputRhom(meshParams.xcells_ini, 0.0);
```

Comment:
And now if we revert the xcells_ini etc back into the Parameters struct, the changes to these project files (bar the FSgrid templating) can be reverted.